packages/node_modules/@webex/plugin-meetings/src/members/index.js

```diff
@@ -36,58 +40,179 @@ export default class Members extends StatelessSparkPlugin {
+        }
+        if (payload.oldSelf) {
+          oldSelfId = payload.oldSelf.id;
+        }
       }
-    });
-    Events.on('LOCUS_INFO_DELTA_PARTICIPANTS', (payload) => {
-      if (payload) {
-        this.handleLocusInfoUpdatedParticipants(payload);
+      if (newSelfId) {
+        const theSelf = this.membersCollection.get(newSelfId);
+        if (theSelf) {
+          theSelf.setIsSelf(true);
+        }
       }
+      if (oldSelfId) {
+        const notSelf = this.membersCollection.get(oldSelfId);
+        if (notSelf) {
+          notSelf.setIsSelf(false);
+        }
+      }
+      this.selfId = newSelfId;
+      this.trigger(EVENT_TRIGGERS.MEMBERS_SELF_UPDATE, {
+        activeSelfId: newSelfId,
+        endedSelfId: oldSelfId
+      });
     });
   }
 
   /**
    * @returns {undefined}
    */
-  setUpLocusUrlListener() {
-    Events.on(EVENTS.LOCUS_URL_UPDATE, (payload) => {
+  setUpLocusHostListener() {
+    Events.on(EVENTS.LOCUS_INFO_UPDATE_HOST, (payload) => {
+      let newHostId = null;
+      let oldHostId = null;
       if (payload) {
-        this.setLocusUrl(null, payload);
+        if (payload.newHost) {
+          newHostId = payload.newHost.id;
+        }
+        if (payload.oldHost) {
+          oldHostId = payload.oldHost.id;
+        }
       }
+      if (newHostId) {
+        const theHost = this.membersCollection.get(newHostId);
+        if (theHost) {
+          theHost.setIsHost(true);
+        }
+      }
+      if (oldHostId) {
+        const notHost = this.membersCollection.get(oldHostId);
+        if (notHost) {
+          notHost.setIsHost(false);
+        }
+      }
+      this.hostId = newHostId;
+      this.trigger(EVENT_TRIGGERS.MEMBERS_HOST_UPDATE, {
+        activeHostId: newHostId,
+        endedHostId: oldHostId
+      });
     });
   }
 
   /**
-   * @param {Object} payload
-   * @returns {Object}
+   * @returns {undefined}
    */
-  handleLocusInfoUpdatedParticipants(payload) {
-    const membersUpdate = this.membersCollection.update(payload);
+  setUpLocusParticipantsListener() {
+    Events.on(EVENTS.LOCUS_INFO_FULL_PARTICIPANTS, (payload) => {
+      this.handleParticipantsPayload(payload);
+    });
+    Events.on(EVENTS.LOCUS_INFO_DELTA_PARTICIPANTS, (payload) => {
+      this.handleParticipantsPayload(payload);
+    });
+  }
+
+  handleParticipantsPayload(payload) {
+    if (payload) {
+      const delta = this.handleLocusInfoUpdatedParticipants(payload);
+      const full = this.handleMembersUpdate(delta); // SDK should propagate the full list for both delta and non delta updates
+      this.trigger(EVENT_TRIGGERS.MEMBERS_UPDATE, {delta, full});
+    }
+  }
+
+  handleMembersUpdate(membersUpdate) {
     if (membersUpdate) {
-      this.trigger('MEMBERS_UPDATE', membersUpdate);
+      if (membersUpdate.updated) {
+        this.constructMembers(membersUpdate.updated);
+      }
+      if (membersUpdate.added) {
+        this.constructMembers(membersUpdate.added);
+      }
```

Comment:
is there a deleted case?

Reply:
Those go into updated, I will add comments, once added to the list, locus participants are never "removed" they are only updated with isNotInMeeting